unlock-app/src/__tests__/services/walletService.test.js

```diff
@@ -662,73 +662,62 @@ describe('WalletService', () => {
 
     describe('signData', () => {
       const account = '0x123'
-      const data = 'please sign me'
+      let data = 'please sign me'
 
-      it('should call first eth.personal.sign', done => {
-        expect.assertions(4)
-        walletService.web3.eth.personal.sign = jest.fn(
-          (dataToSign, signer, callback) => {
-            expect(dataToSign).toEqual(data)
-            expect(signer).toEqual(account)
-            return callback(null /* error */, 'signature')
-          }
-        )
-        walletService.signData(account, data, (error, signature) => {
-          expect(walletService.web3.eth.personal.sign).toHaveBeenCalled()
-          expect(signature).toEqual('signature')
-          done()
+      describe('if the provider is metamask', () => {
+        it('should use eth_signTypedData_v3 and stringify the data', done => {
+          expect.assertions(2)
+          data = []
+          walletService.web3.currentProvider.isMetaMask = true
+          walletService.web3.currentProvider.send = jest.fn((args, cb) => {
+            expect(args.method).toBe('eth_signTypedData_v3')
+            expect(args.params[1]).toBe(JSON.stringify(data))
+            return cb(null, { result: '' })
+          })
+          walletService.signData(account, data, () => {
+            done()
+          })
         })
       })
 
-      it('should call web3.eth.sign if eth.personal.sign fails', done => {
-        expect.assertions(7)
-
-        walletService.web3.eth.personal.sign = jest.fn(
-          (dataToSign, signer, callback) => {
-            expect(dataToSign).toEqual(data)
-            expect(signer).toEqual(account)
-            const error = {}
-            return callback(error, null /* signature */)
-          }
-        )
-
-        walletService.web3.eth.sign = jest.fn(
-          (dataToSign, signer, callback) => {
-            expect(dataToSign).toEqual(data)
-            expect(signer).toEqual(account)
-            return callback(null /* error */, 'signature')
-          }
-        )
-
+      it('should send the the method to the provider with the right params and yield the signature', done => {
```

Comment:
How would you feel about update the description of the test to highlight that this is the path that a non-metamask provider should go through?